src/twitter/TwitterProfile.js

```diff
@@ -0,0 +1,170 @@
+      rooms.forEach(entry => {
+        intent.setRoomName(entry.matrix.getId(), util.formatStringFromObject(ROOMNAME_FORMAT, new_profile));
+      });
+    }
+
+    if(update_description) {
+      rooms.forEach(entry => {
+        intent.setRoomTopic(entry.matrix.getId(), util.formatStringFromObject(ROOMTOPIC_FORMAT, new_profile));
+      });
+    }
+
+    var url = null;
+    if(update_avatar) {
+      //We have to replace _normal because it gives us a bad quality image
+      // E.g https://pbs.twimg.com/profile_images/796729706318012418/VdozW4mO_normal.jpg
+      // becomes https://pbs.twimg.com/profile_images/796729706318012418/VdozW4mO.jpg
+      const image_url = new_profile.profile_image_url_https.replace("_normal", "");
+      log.verbose(`Updating avatar for @${new_profile.screen_name} with @${image_url}.`);
+      url = yield util.uploadContentFromUrl(this._twitter.bridge, image_url);
+      url = url.mxc_url;
+      intent.setAvatarUrl(url);
+      rooms.forEach(entry => {
+        intent.setRoomAvatar(entry.matrix.getId(), url);
+      });
+    }
+
+    return this._twitter.storage.cache_user_profile(new_profile.id_str, new_profile.screen_name, new_profile, ts);
+  }
+
+  format_displayname (profile) {
+    let fmt = DISPLAYNAME_FORMAT;
+    if (this._config.formatting) {
+      if (this._config.formatting.user_displayname) {
+        fmt = this._config.formatting.user_displayname;
+      }
+    }
+    return util.formatStringFromObject(fmt, profile);
+  }
+
+  /**
+   * Get a Twitter profile by a users Twitter ID.
+   *
+   * @param  {number} id Twitter Id
+   * @return {Promise<TwitterProfile>} A promise containing the Twitter profile
+   * to be returned. See https://dev.twitter.com/rest/reference/get/users/show
+   */
+  get_by_id (twitter_id) {
+    log.info("Looking up T" + twitter_id);
+    return this._storage.get_profile_by_id(twitter_id).then((profile)=>{
+      if(profile != null) {
+        if(!profile._outofdate) {
+          return profile;
+        }
+      }
+      return this._get_profile({twitter_id});
+    });
+  }
+
+  /**
+   * Get a Twitter profile by a users screen name.
+   * @param  {number} id Twitter Screen name
+   * @return {Promise<TwitterProfile>} A promise containing the Twitter profile
+   * to be returned. See {@link}
+   *
+   * @see {@link https://dev.twitter.com/rest/reference/get/users/show}
+   */
+  get_by_screenname (screen_name) {
+    log.info("Looking up T" + screen_name);
+    return this._twitter.storage.get_profile_by_name(screen_name).then((profile)=>{
+      if(profile != null) {
+        if(!profile._outofdate) {
+          return profile;
+        }
+      }
+      return this._get_profile({screen_name});
+    });
+  }
+
+  _get_profile (data) {
+    return this._twitter.client_factory.get_client().then(client => {
+      return client.getAsync('users/show', data);
+    }).then(user => {
+      return this.update_profile(user).thenReturn(user);
+    }).catch(error => {
```

Comment:
This catch should really be on `client.getAsync(..` above otherwise the log might lie.